Server/src/redis/MsgBusImpl_redis.cpp

```diff
@@ -0,0 +1,232 @@
+    redisMgr_.ExitRedisManager();
+}
+
+/**
+ * Reset all Tables once FRR reconnects to BMP, this will not disable table population
+ *
+ * \param [in] N/A
+ */
+void MsgBusImpl_redis::ResetAllTables() {
+    redisMgr_.ResetAllTables();
+}
+
+/**
+ * Abstract method Implementation - See MsgBusInterface.hpp for details
+ */
+void MsgBusImpl_redis::update_Peer(obj_bgp_peer &peer, obj_peer_up_event *up, obj_peer_down_event *down, peer_action_code code) {
+
+    // Below attributes will be populated if exists, and no matter bgp neighbor is up or down
+    vector<swss::FieldValueTuple> fieldValues;
+    fieldValues.reserve(MAX_ATTRIBUTES_COUNT);
+    vector<string> keys;
+    keys.emplace_back(peer.peer_addr);
+
+    fieldValues.emplace_back(make_pair("peer_addr", peer.peer_addr));
+    fieldValues.emplace_back(make_pair("peer_asn", to_string(peer.peer_as)));
+    fieldValues.emplace_back(make_pair("peer_rd", peer.peer_rd));
+    fieldValues.emplace_back(make_pair("remote_port", to_string(up->remote_port)));
+    fieldValues.emplace_back(make_pair("local_asn", to_string(up->local_asn)));
+    fieldValues.emplace_back(make_pair("local_ip", up->local_ip));
+    fieldValues.emplace_back(make_pair("local_port", to_string(up->local_port)));
+    fieldValues.emplace_back(make_pair("sent_cap", up->sent_cap));
+    fieldValues.emplace_back(make_pair("recv_cap", up->recv_cap));
+
+    switch (code) {
+        case PEER_ACTION_DOWN:
+        {
+            // PEER DOWN only
+            fieldValues.emplace_back(make_pair("bgp_err_code", to_string(down->bgp_err_code)));
+            fieldValues.emplace_back(make_pair("bgp_err_subcode", to_string(down->bgp_err_subcode)));
+            fieldValues.emplace_back(make_pair("error_text", down->error_text));
+
+        }
+        break;
+    }
+    redisMgr_.WriteBMPTable(BMP_TABLE_NEI, keys, fieldValues);
+}
+
+
+/**
+ * Abstract method Implementation - See MsgBusInterface.hpp for details
+ */
+void MsgBusImpl_redis::update_unicastPrefix(obj_bgp_peer &peer, vector<obj_rib> &rib,
+                                        obj_path_attr *attr, unicast_prefix_action_code code) {
+    if (attr == NULL)
+        return;
+
+    // Loop through the vector array of rib entries
+    vector<swss::FieldValueTuple> addFieldValues;
+    addFieldValues.reserve(MAX_ATTRIBUTES_COUNT);
+    vector<string> del_keys;
+    string neigh = peer.peer_addr;
+
+    for (size_t i = 0; i < rib.size(); i++) {
+        // rib table schema as BGP_RIB_OUT_TABLE|192.181.168.0/25|10.0.0.59
+        vector<string> keys;
+        string redisMgr_pfx = rib[i].prefix;
+        redisMgr_pfx += "/";
+        redisMgr_pfx += to_string(rib[i].prefix_len);
+        keys.reserve(MAX_ATTRIBUTES_COUNT);
+        keys.emplace_back(redisMgr_pfx);
+        keys.emplace_back(peer.peer_addr);
+
+        switch (code) {
+
+            case UNICAST_PREFIX_ACTION_ADD:
+            {
+                addFieldValues.emplace_back(make_pair("origin", attr->origin));
```

Comment:
The addFieldValues is defined outside the loop, and there is no reset or clear operation at the beginning or end of each loop iteration.
As a result, addFieldValues will continue to accumulate values. Is this behavior intentional?